pkg/app/bootstrap.go

```diff
@@ -112,56 +107,70 @@ func bootstrap(clicontext *cli.Context) error {
 }
 
 func registryCA(clicontext *cli.Context) error {
-	ca := clicontext.String("registry-ca-keypair")
-	if len(ca) == 0 {
-		return nil
-	}
-	mirror := clicontext.String("dockerhub-mirror")
-	if len(mirror) == 0 {
-		return errors.New("`registry-ca-keypair` should be used with `dockerhub-mirror`")
-	}
+	// Loop over all registry strings
+	if clicontext.IsSet("registry") {
+		for _, reg := range strings.Split(clicontext.String("registry"), ";") {
+			// Split into parts
+			parts := strings.Split(reg, ",")
+			if len(parts) != 4 {
+				return fmt.Errorf("invalid registry string: %s", reg)
+			}
 
-	// parse ca/key/cert
-	kvPairs := strings.Split(ca, ",")
-	if len(kvPairs) != 3 {
-		return errors.New("`registry-ca-keypair` requires ca/key/cert 3 part separated by ','")
-	}
-	names := []string{"ca", "cert", "key"}
-	for _, pair := range kvPairs {
-		kv := strings.SplitN(pair, "=", 2)
-		index := -1
-		for i, name := range names {
-			if name == kv[0] {
-				index = i
-				break
+			// Parse registry and ca/key/cert parts
+			names := []string{"registry", "ca", "key", "cert"}
+			for _, part := range parts {
+				kv := strings.SplitN(part, "=", 2)
+				if len(kv) != 2 {
+					return fmt.Errorf("invalid part: %s", part)
+				}
+
+				// Check for valid key
+				index := -1
+				for i, name := range names {
+					if name == kv[0] {
+						index = i
+						break
+					}
+				}
+				if index == -1 {
+					return errors.Newf("parse error: `%s` is not a valid ca/key/cert key or it's duplicated")
+				}
+
+				// Read file if not registry part
+				if kv[0] != "registry" {
+					exist, err := fileutil.FileExists(kv[1])
+					if err != nil {
+						return errors.Wrap(err, fmt.Sprintf("failed to parse file path %s", part))
+					}
+					if !exist {
+						return fmt.Errorf("file %s doesn't exist", kv[1])
+					}
+
+					// Generate file path
+					path, err := fileutil.ConfigFile(fmt.Sprintf("registry_%s.pem", kv[0]))
```

Comment:
I am curious why do we have to read and write the pem files? can't we just use the original one?

Reply:
This is copied from the original implementation. The main purpose is to make sure these 3 files are in the same directory since we will mount this directory to the buildkitd container.

Reply:
So we basically copy 3 pems to the new dir? If that's the case, do we still need binding to allow the permission to `/etc` or other folders? I think buildkit will have permission to the new dir?

Reply:
What do you mean the original impl? and Why not just use `io.Copy` instead of `read` and `write`?

Reply:
> Why not just use `io.Copy` instead of `read` and `write`?

That should be better.

> So we basically copy 3 pems to the new dir? If that's the case, do we still need binding to allow the permission to `/etc` or other folders? I think buildkit will have permission to the new dir?

Need to mount a directory to the buildkitd container. This directory is used to store the envd config, which should be safe.

This PR supports multiple registries which means it needs different CAs for different registries. It might need to differentiate the file names.